rrdd/rrdd_server.ml

```diff
@@ -576,17 +575,17 @@ module Plugin = struct
     (* Read, parse, and combine metrics from all registered plugins. *)
     let read_stats () : (Rrd.ds_owner * Ds.ds) list =
       let plugins = Mutex.execute registered_m
-          (fun _ -> Hashtblext.to_list registered) in
+          (fun _ -> Hashtbl.to_seq registered) in
       let process_plugin acc (uid, plugin) =
         try
           let payload = get_payload ~uid plugin in
           List.rev_append payload.Rrd_protocol.datasources acc
         with _ -> acc
       in
-      List.iter decr_skip_count plugins;
+      Seq.iter decr_skip_count plugins;
```

Comment:
you're safe to use `Seq` here because iter is strict:

```
val iter : ('a -> unit) -> 'a t -> unit

Iterate on the sequence, calling the (imperative) function on every element. The traversal happens immediately and will not terminate on infinite sequences.
```

you would probably be fine below as well, but personally i would stick to list here

Reply:
process_plugin has side effects, but since the order doesn't change and it happens after iter has finished anyway, I don't see the issue.

What are your fears in this case?